src/backend/api/meals.js

```diff
@@ -1,15 +1,130 @@
 const express = require("express");
+const app = require("../app");
+//const { max } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
+//Get meals that has a price smaller than maxPrice
 router.get("/", async (request, response) => {
+  console.log("akjdskank");
+  if ("maxPrice" in request.query) {
+    const dataTable = await knex("meal").where(
+      "price",
+      "<",
+      Number(request.query.maxPrice)
+    );
+    return response.json(dataTable);
+  }
+
+  //Get meals that still has available reservations
+  if ("availableReservation" in request.query) {
+    const availableReservations = await knex("meal")
+      .join("reservation", "reservation.meal_id", "=", "meal.id")
+      .where("max_reservations", ">", "number_of_guests");
+    response.json(availableReservations);
+  }
+  // Get meals that partially match a title
+  if ("title" in request.query) {
+    const getTitle = await knex("meal").where("title", "like", "%pasta%");
+    response.json(getTitle);
+  }
+  //Get meals that has been created after the date
+  // if (createdAfter in request.query) {
+  //   const getCreatedDate = await knex("meal").where(function () {
+  //     this.where("created_date", ">", "2022-01-20");
+  //   });
+  //   response.json(getCreatedDate);
+  // }
+  if ("createdAfter" in request.query) {
+    const createdAfter = await knex("meal").where(
+      "created_date",
+      ">",
+      "2022-01-20" //we can also give request.query.createdAfter
+    );
+
+    if (createdAfter.length === 0) {
+      return response
+        .status(404)
+        .json({ Error: "No meal found created this date" });
+    }
+    return response.json(createdAfter);
```

Comment:
Great, that you give the user some feedback if there is no meal that was created after the date but remember it is not an error right? so you don't want to give a status(404) 
You can just send a string that says **No meal found created this date** so you'll have something like res,send('your message here')